runtime/src/utils/buffer/pool.rs

```diff
@@ -220,7 +222,7 @@ impl PoolRef {
 
         // Cache the result in the buffer pool.
         let Ok(page_buf) = fetch_result else {
-            return Err(Error::ReadFailed);
+            return Err(Error::ReadFailed(Box::new(fetch_result.unwrap_err())));
```

Comment:
I think using `match` rather than `if/else` might be cleaner here (?)

Reply:
🤷 I generally prefer  "let else" if there's only two clauses and one of them is err.